### PR TITLE
DB/Loot - Fix dropchance of [A Letter to Yvette]

### PR DESCRIPTION
**Pull request by schell244**
## Changes Proposed:
-  drop chances according to https://classic.wowhead.com/item=2839
- add drop chance to some additional npc's 
-  <strike>add missing condition: Horde only drop</strike>

<strike>Condition Horde only is a "custom fix", on classic the item seems to drop for Alliance player as well, but they cannot start the quest.</strike>
https://classic.wowhead.com/item=2839/a-letter-to-yvette#comments
_I actually found this as my Night Elf druid (33 at the time) west of Brill. Was kind of funny, wish I was eligible for the quest...told me I wasn't the correct race._

## Issues Addressed:
-  

## SOURCE:
https://classic.wowhead.com/item=2839

## Tests Performed:
-  insert sql works, tested ingame with both factions

## How to Test the Changes:
1. farm item [A Letter to Yvette] with both Horde and Alliance characters
2. it should only drop for both Horde and Alliance

## Known Issues and TODO List:
- 

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Kitzunu** [comment]
While I appreciate your contribution we got two issues.

1. We do not use Replace Into. Use either DELETE/INSERT or UPDATE.
2. Do not add custom fixes, as this also happened on retail 3.3.5a as far as I can tell, it should be that way here too.

**Comment by schell244** [comment]
Thanks for your feedback, I've updated the PR accordingly.

File: data/sql/updates/pending_db_world/rev_1637094877418094500.sql [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+INSERT INTO `creature_loot_template` (`Entry`, `Item`, `Reference`, `Chance`, `QuestRequired`, `LootMode`, `GroupId`, `MinCount`, `MaxCount`, `Comment`) VALUES
+(1537,2839,0,0.01,0,1,0,1,1,'Scarlet Zealot - A Letter to Yvette');
+
+-- Rotting Ancestor
+INSERT INTO `creature_loot_template` (`Entry`, `Item`, `Reference`, `Chance`, `QuestRequired`, `LootMode`, `GroupId`, `MinCount`, `MaxCount`, `Comment`) VALUES
+(1530,2839,0,0.01,0,1,0,1,1,'Rotting Ancestor - A Letter to Yvette');
+
+-- Wandering Spirit
+INSERT INTO `creature_loot_template` (`Entry`, `Item`, `Reference`, `Chance`, `QuestRequired`, `LootMode`, `GroupId`, `MinCount`, `MaxCount`, `Comment`) VALUES
+(1532,2839,0,0.01,0,1,0,1,1,'Wandering Spirit - A Letter to Yvette');
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1637094877418094500.sql:33]: Merge these lines into 1

**Review by Nefertumm — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1637094877418094500.sql [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+-- Rattlecage Soldier, Cracked Skull Soldier
+UPDATE `creature_loot_template` SET `Chance`=3 WHERE `Item`=2839 AND `Entry` IN (1520,1523);
+
+INSERT INTO `creature_loot_template` (`Entry`, `Item`, `Reference`, `Chance`, `QuestRequired`, `LootMode`, `GroupId`, `MinCount`, `MaxCount`, `Comment`) VALUES
+(1664,2839,0,0.12,0,1,0,1,1,'Captain Vachon - A Letter to Yvette'),
+(1770,2839,0,0.05,0,1,0,1,1,'Moonrage Darkrunner - A Letter to Yvette'),
+(1548,2839,0,0.01,0,1,0,1,1,'Cursed Darkhound - A Letter to Yvette'),
+(1537,2839,0,0.01,0,1,0,1,1,'Scarlet Zealot - A Letter to Yvette'),
+(1530,2839,0,0.01,0,1,0,1,1,'Rotting Ancestor - A Letter to Yvette'),
+(1532,2839,0,0.01,0,1,0,1,1,'Wandering Spirit - A Letter to Yvette');
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1637094877418094500.sql:17]: add a new line at the end please.

**Review by Nefertumm — APPROVED** [review]

**Review by Kitzunu — CHANGES_REQUESTED** [review]
always delete before insert

**Comment by claudiodfc** [comment]
### Test result: ✔
- Low chance drop for all NPCs.
- Killed 300 `Moonrage Darkrunner` and none dropped the item.
- Killed 60 `Darkeye Bonecasters` and dropped the quest item.
- Drops for Alliance but they can't start the quest `That quest isn't available for your race`.

**Review by claudiodfc — APPROVED** [review]

**Review by Winfidonarleyan — APPROVED** [review]

**Pull request merged**